### PR TITLE
update to latest HEAD hash on master

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
@peastman I think these errors are what is causing the windows build to fail 
```
2022-06-10T19:03:08.1934522Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(255): error C3861: 'min': identifier not found
2022-06-10T19:03:08.1939976Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(255): error C3861: 'max': identifier not found
```
Some quick googling makes it looks like it could be a missing header file, but it seems to be building fine on GHA in the openmm repo, I checked and it is using the same vs code runtime, any ideas? Or is there a different error that I'm missing?

I could try adding a patch to include `#include<minmax.h>` here:
https://github.com/openmm/openmm/blob/master/platforms/reference/src/ReferenceTabulatedFunction.cpp#L38

to see if that fixes it, but I'm not sure still why it works on the main repo CI but seems to fail here.

**Comment by peastman** [comment]
I think I know what the problem is.  At the very top of ReferenceTabulatedFunction.cpp, we need to add

```c++
#ifdef _MSC_VER
    // Prevent Windows from defining macros that interfere with other code.
    #define NOMINMAX
#endif

```

It's a strange historical quirk of Visual Studio.  Way back in ancient history, Microsoft decided one of their header files would `#define` functions for `min()` and `max()`.  Then they got added to the standard C/C++ headers, but as regular functions instead of preprocessor macros.  To maintain backward compatibility, Microsoft kept including their versions in place of the standard versions.  If you want to force it to provide the standard versions of those functions, you need to define the symbol `NOMINMAX` before including the header.  We do that in most other places that `#include <cmath>`, but it looks like it got missed here.

**Comment by mikemhenry** [comment]
> I think I know what the problem is. At the very top of ReferenceTabulatedFunction.cpp, we need to add
> 
> ```c++
> #ifdef _MSC_VER
>     // Prevent Windows from defining macros that interfere with other code.
>     #define NOMINMAX
> #endif
> ```
> 
> It's a strange historical quirk of Visual Studio. Way back in ancient history, Microsoft decided one of their header files would `#define` functions for `min()` and `max()`. Then they got added to the standard C/C++ headers, but as regular functions instead of preprocessor macros. To maintain backward compatibility, Microsoft kept including their versions in place of the standard versions. If you want to force it to provide the standard versions of those functions, you need to define the symbol `NOMINMAX` before including the header. We do that in most other places that `#include <cmath>`, but it looks like it got missed here.

Wow what a headache! Do we want to just fix this upstream or with a patch here?

**Comment by peastman** [comment]
We should fix it in OpenMM.  It's just an oversight that it got left out.  I wonder why CI didn't catch it though?  The calls to `min()` and `max()` were added back in April, and Windows builds are still passing.

**Comment by mikemhenry** [comment]
Cool, I can make a PR.

And yah, that is what made this one a tough one to track down (at least with my minimal windows system programming experience) I was hoping that it would be something like a different MSV runtime or something.

**Comment by Tobias-Fischer** [comment]
See https://github.com/conda-forge/gnuradio-soapy-feedstock/blob/3a39d3997e21e281b6f679090d4db24eef48f0f6/recipe/bld.bat#L5-L6 re the nominmax stuff.

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Hmm .. maybe try https://github.com/conda-forge/cartographer-feedstock/blob/9940311bd2c507b11d22d864d162247d41d1c21b/recipe/bld.bat#L4 instead?

**Comment by peastman** [comment]
We may be misdiagnosing the problem.  The build log contains the message

```
%SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(33): warning C4005: 'NOMINMAX': macro redefinition
%SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp: note: see previous definition of 'NOMINMAX'
```

So clearly we're successfully defining the macro (twice!).  I think I see what might be the real problem:

```c++
#ifdef _MSC_VER

#if _MSC_VER < 1800
/**
 * We need to define this ourselves, since Visual Studio is missing round() from cmath.
 */
static int round(double x) {
    return (int) (x+0.5);
}
#else
#include <cmath>
#endif  // MSC_VER < 1800


#else
#include <cmath>
#endif
```

Notice that if `_MSC_VER` is defined and is less than 1800, `cmath` never gets included.  I think this is what we want to do:

```c++
#include <cmath>

#ifdef _MSC_VER
#if _MSC_VER < 1800
/**
 * We need to define this ourselves, since Visual Studio is missing round() from cmath.
 */
static int round(double x) {
    return (int) (x+0.5);
}
#endif
#endif
```

Does that look right?

**Comment by Tobias-Fischer** [comment]
I'm not sure. The errors are clearly related to min/max:
```

2022-06-21T20:38:09.5461852Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(259): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5470115Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(259): error C3861: 'max': identifier not found
2022-06-21T20:38:09.5476984Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(282): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5480118Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(282): error C3861: 'max': identifier not found
2022-06-21T20:38:09.5505119Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(283): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5512906Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(283): error C3861: 'max': identifier not found
2022-06-21T20:38:09.5514646Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(307): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5517439Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(307): error C3861: 'max': identifier not found
2022-06-21T20:38:09.5527941Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(308): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5548999Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(308): error C3861: 'max': identifier not found
2022-06-21T20:38:09.5575080Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(309): error C3861: 'min': identifier not found
2022-06-21T20:38:09.5577544Z %SRC_DIR%\platforms\reference\src\ReferenceTabulatedFunction.cpp(309): error C3861: 'max': identifier not found
```

I'd in https://github.com/openmm/openmm/blob/master/platforms/reference/src/ReferenceTabulatedFunction.cpp
1) include <algorithm>
2) change `min` to `std::min` explicitly and same for `max`

**Comment by peastman** [comment]
> change min to std::min explicitly and same for max

We already have the line

```c++
using namespace std;
```

**Comment by Tobias-Fischer** [comment]
> > change min to std::min explicitly and same for max
> 
> We already have the line
> 
> ```c++
> using namespace std;
> ```

I wouldn't trust it ;).

**Comment by peastman** [comment]
> I wouldn't trust it ;).

Only if the compiler has decided to ignore the C++ spec! :)  Anyway, adding `#include <algorithm>` seems like a good first thing to try.

**Comment by mikemhenry** [comment]
Okay PR here: https://github.com/openmm/openmm/pull/3658
Once it gets merged in I can re-do the build here with it.

**Comment by peastman** [comment]
Thanks!  I'll merge it as soon as CI finishes.

File: recipe/build_openmm.bat [hunk 1]
```diff
@@ -5,6 +5,11 @@ cd build
 
 set "CUDA_TOOLKIT_ROOT_DIR=%CUDA_PATH:\=/%"
 
+:: define NOMINMAX since openmm headers expect min/max to be functions not macros
+set "CFLAGS=%CFLAGS% -DNOMINMAX"
+set "CXXFLAGS=%CXXFLAGS% -DNOMINMAX"
```
Comment (peastman) [inline recipe/build_openmm.bat:10]: We should be able to remove these lines now.  That turned out not to be the problem.

Comment (mikemhenry) [reply recipe/build_openmm.bat]: Done!

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Pull request merged**